### DISCUSSION TITLE
Add plain javascript in all pages

### DISCUSSION THREAD

**Discussion by salahy** [blitz-js/blitz#2927] [General]
Hello,
I am trying to add shared script in all pages, not sure what is the best way to do it, I tried to add it on document file but in the doc it says don't add it there.

Thank you

**Comment by beerose** [comment]
Hey,

There's a `Script` component that you can use: https://blitzjs.com/docs/script-component. If you want to add it to all the pages you can use `DocumentHead` (instead of `Head`) in the `_document` file.

**Reply by salahy** [reply]
Hi,
Thank you for your answer, I already tried that but I am getting 
`ReferenceError: document is not defined` when I add to DocumentHead, I am trying simple javascript code to test it, but if I add this script in all pages it works

**Reply by beerose** [reply]
This error usually happens when the code runs on the server, while `document` is only available in the browser. Do you get this error from your code or from some Blitz internals?

**Reply by salahy** [reply]
This is example of the code
```
import { Document, Html, DocumentHead, Main, BlitzScript} from "blitz"
import { Header } from "core/components/Header"
import { Footer } from "core/components/Footer"
import { Script } from "blitz"
class MyDocument extends Document {
  render() {
    return (
      <Html>
        <DocumentHead>
          <Script>{`alert("hello")`}</Script>
        </DocumentHead>
        <body>
          <Header />
          <Main />
          <BlitzScript />
          <Footer />
        </body>
      </Html>
    )
  }
}
export default MyDocument
```


This will not trigger the alert, if I add  strategy="beforeInteractive" to the script, it will show the error page

```
ReferenceError: document is not defined
This error happened while generating the page. Any console logs will be displayed in the terminal window.
```

**Reply by beerose** [reply]
Ok, I'm sorry, I was wrong about the `_document` file. I missed the note in the docs:
```
<Script> must not be placed in either a <Head> component or in _document.js.
```
So you can either put it in all the pages, or in a layout component, something like:
```
const Layout = ({ title, children }: LayoutProps) => {
  return (
    <>
      <Script>{`alert("hello")`}</Script>
      {children}
    </>
  )
}
```
